tofhir-engine/src/main/scala/io/tofhir/engine/mapping/FhirMappingJobManager.scala

```diff
@@ -174,29 +174,29 @@ class FhirMappingJobManager(
   /**
    * Runnable for scheduled periodic mapping job
    *
-   * @param id                 Job identifier
+   * @param job                Fhir mapping job
    * @param startTime          Initial start time for source data
    * @param tasks              Mapping tasks
    * @param sourceSettings     Settings for the source system
    * @param sinkSettings       FHIR sink settings/configurations
    * @param schedulingSettings Scheduling information
    * @return
    */
-  private def runnableMappingJob(id: String,
+  private def runnableMappingJob(job: FhirMappingJob,
                                  startTime: LocalDateTime,
                                  tasks: Seq[FhirMappingTask],
                                  sourceSettings: Map[String, DataSourceSettings],
                                  sinkSettings: FhirSinkSettings,
                                  terminologyServiceSettings: Option[TerminologyServiceSettings] = None,
                                  identityServiceSettings: Option[IdentityServiceSettings] = None,
                                  schedulingSettings: SchedulingSettings): Future[Unit] = {
-    val timeRange = getScheduledTimeRange(id, mappingJobScheduler.get.folderUri, startTime)
+    val timeRange = getScheduledTimeRange(job.id, mappingJobScheduler.get.folderUri, startTime)
     logger.info(s"Running scheduled job with the expression: ${schedulingSettings.cronExpression}")
     logger.info(s"Synchronizing data between ${timeRange._1} and ${timeRange._2}")
-    val mappingJobExecution = FhirMappingJobExecution(jobId = id, mappingTasks = tasks)
+    val mappingJobExecution = FhirMappingJobExecution(job = job, mappingTasks = tasks)
     executeMappingJob(mappingJobExecution, sourceSettings, sinkSettings, terminologyServiceSettings, identityServiceSettings, Some(timeRange))
       .map(_ => {
-        val writer = new FileWriter(s"${mappingJobScheduler.get.folderUri.getPath}/$id.txt", true)
+        val writer = new FileWriter(s"${mappingJobScheduler.get.folderUri.getPath}/$job.txt", true)
```

Comment:
Is this change intentional? Do we have test case covering this part?

Reply:
@Okanmercan99 can you please add a test case covering this (to be implemented in another branch, resolving this issue for now to merge the main feature)